File: AppDev_Act8/app/src/main/java/com/example/appdev_act8/MainActivity.java
package com.example.appdev_act8;

import android.content.DialogInterface;
import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ListView;

import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    EditText searchBox, prodName, priceInp;
    Button saveBtn;
    ListView listView;

    ArrayList<String> itemList = new ArrayList<>();
    ArrayAdapter<String> arrayAdapter;

    private View add;

    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        searchBox = findViewById(R.id.searchBox);
        prodName = findViewById(R.id.prodName);
        priceInp = findViewById(R.id.priceInp);
        saveBtn = findViewById(R.id.saveBtn);
        listView = findViewById(R.id.listView);

        itemList = PriceHelper.readData(this);
        arrayAdapter = new ArrayAdapter<>(this, android.R.layout.simple_list_item_1,
                android.R.id.text1, itemList);

        listView.setAdapter(arrayAdapter);

        searchBox.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {
            }

            @Override
            public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
                // Filter the list based on the user's input
                arrayAdapter.getFilter().filter(charSequence.toString());
            }

            @Override
            public void afterTextChanged(Editable editable) {
            }
        });

        saveBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String prod = prodName.getText().toString();
                String price = priceInp.getText().toString();

                // Check if both product name and price are not empty
                if (!prod.isEmpty() && !price.isEmpty()) {
                    // Concatenate the product name and price
                    String result = prod + " ₱" + price;

                    // Add the result to the item list
                    itemList.add(result);

                    // Clear the input fields
                    prodName.setText("");
                    priceInp.setText("");

                    // Write the updated list to the file
                    PriceHelper.writeData(itemList, getApplicationContext());

                    // Notify the adapter that the data has changed
                    arrayAdapter.notifyDataSetChanged();
                }
            }
        });

        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                AlertDialog.Builder alert = new AlertDialog.Builder(MainActivity.this);
                alert.setTitle("REMOVE");
                alert.setMessage("DO YOU WANT REMOVE THIS ITEM TO OUR LIST?");
                alert.setCancelable(false);
                alert.setNegativeButton("No", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        dialog.cancel();
                    }
                });

                alert.setPositiveButton("Yes", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        itemList.remove(position);
                        arrayAdapter.notifyDataSetChanged();
                        PriceHelper.writeData(itemList, getApplicationContext());
                    }
                });

                AlertDialog alertDialog = alert.create();
                alertDialog.show();
            }
        });
    }
}

File: AppDev_Act6/app/src/main/java/com/example/appdev_act6/MainActivity.kt
package com.example.appdev_act6

import android.content.ClipData.Item
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Adapter
import android.widget.AdapterView
import android.widget.AdapterView.OnItemClickListener
import android.widget.GridView
import android.widget.Toast

class MainActivity : AppCompatActivity() , AdapterView.OnItemClickListener{

    private var gridView: GridView? = null
    private var arrayList: ArrayList<item>? = null
    private var adapter: Adapter? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        gridView = findViewById(R.id.my_grid_view)
        arrayList = ArrayList()
        arrayList = setDataList()
        adapter = Adapter(applicationContext, arrayList!!)
        gridView?.adapter = adapter as com.example.appdev_act6.Adapter
        gridView?.onItemClickListener = this
    }

    private fun setDataList(): ArrayList<item> {

        var arrayList: ArrayList<item> = ArrayList()

        arrayList.add(item(R.drawable.yoichi, "YOICHI ISAGI #11"))
        arrayList.add(item(R.drawable.bachira, "MEGURU BACHIRAN #8"))
        arrayList.add(item(R.drawable.rin, "RIN ITOSHI #10"))
        arrayList.add(item(R.drawable.baro, "SHOEI BARO #13"))
        arrayList.add(item(R.drawable.nagi, "SEISHIRO NAGI #7"))
        arrayList.add(item(R.drawable.gin, "GIN GAGAMARU #1"))

        arrayList.add(item(R.drawable.reo, "REO MIKAGE #14"))
        arrayList.add(item(R.drawable.karasu, "TABITO KARASU #6"))
        arrayList.add(item(R.drawable.kenyu, "KENYU YUKIMIYA #5"))
        arrayList.add(item(R.drawable.aryu, "JYUBEI ARYU #2"))
        arrayList.add(item(R.drawable.hiori, "YO HIORI 16"))
        arrayList.add(item(R.drawable.chigiri, "HYOMA CHIGIRI #4"))
        arrayList.add(item(R.drawable.niko, "IKKI NIKO #3"))
        arrayList.add(item(R.drawable.otoya, "EITA OTOYA #9"))

        return arrayList

    }

    override fun onItemClick(parent: AdapterView<*>?, view: View?, position: Int, id: Long) {

        var item: item = arrayList!!.get(position)
        Toast.makeText(applicationContext, item.name,  Toast.LENGTH_LONG).show()
    }
}







